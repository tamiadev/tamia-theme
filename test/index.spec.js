const path = require('path');

const indexPath = '../index';
const cwd = process.cwd();

afterEach(() => {
	process.chdir(cwd);
	jest.resetModules();
});

test('load theme', () => {
	process.chdir(path.resolve(__dirname, 'fixtures'));
	const result = require(indexPath);
	expect(result).toMatchSnapshot();
});

test('throw when theme file not found', () => {
	process.chdir(__dirname);
	const fn = () => require(indexPath);
	expect(fn).toThrow('Theme not found');
});
