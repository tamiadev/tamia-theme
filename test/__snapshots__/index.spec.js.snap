// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load theme 1`] = `
Object {
  "font": "Comic Sans",
  "foo": Object {
    "bar": Object {
      "baz": 42,
    },
  },
  "space": Object {
    "l": "32px",
    "m": "16px",
    "s": "8px",
    "xl": "64px",
    "xs": "4px",
    "xxs": "2px",
  },
}
`;
